src/main/java/net/earthcomputer/clientcommands/command/arguments/FormattedTextArgumentType.java

```diff
@@ -0,0 +1,226 @@
+                .put("dark_green", new Styler((s, o) -> s.withFormatting(Formatting.DARK_GREEN), 0))
+                .put("dark_purple", new Styler((s, o) -> s.withFormatting(Formatting.DARK_PURPLE), 0))
+                .put("dark_red", new Styler((s, o) -> s.withFormatting(Formatting.DARK_RED), 0))
+                .put("gold", new Styler((s, o) -> s.withFormatting(Formatting.GOLD), 0))
+                .put("gray", new Styler((s, o) -> s.withFormatting(Formatting.GRAY), 0))
+                .put("green", new Styler((s, o) -> s.withFormatting(Formatting.GREEN), 0))
+                .put("italic", new Styler((s, o) -> s.withFormatting(Formatting.ITALIC), 0))
+                .put("light_purple", new Styler((s, o) -> s.withFormatting(Formatting.LIGHT_PURPLE), 0))
+                .put("obfuscated", new Styler((s, o) -> s.withFormatting(Formatting.OBFUSCATED), 0))
+                .put("red", new Styler((s, o) -> s.withFormatting(Formatting.RED), 0))
+                .put("reset", new Styler((s, o) -> s.withFormatting(Formatting.RESET), 0))
+                .put("strikethrough", new Styler((s, o) -> s.withFormatting(Formatting.STRIKETHROUGH), 0))
+                .put("underline", new Styler((s, o) -> s.withFormatting(Formatting.UNDERLINE), 0))
+                .put("white",  new Styler((s, o) -> s.withFormatting(Formatting.WHITE), 0))
+                .put("yellow", new Styler((s, o) -> s.withFormatting(Formatting.YELLOW), 0))
+
+                .put("font", new Styler((s, o) -> s.withFont(Identifier.tryParse(o.get(0))), 1, new String[]{"alt", "default"}))
+                .put("hex", new Styler((s, o) -> s.withColor(TextColor.fromRgb(Integer.parseInt(o.get(0), 16))), 1))
+                .put("insert", new Styler((s, o) -> s.withInsertion(o.get(0)), 1))
+
+                .put("click", new Styler((s, o) -> s.withClickEvent(new ClickEvent(ClickEvent.Action.byName(o.get(0)), o.get(1))), 2, new String[]{"change_page", "copy_to_clipboard", "open_file", "open_url", "run_command", "suggest_command"}))
+                .put("hover", new Styler((s, o) -> s.withHoverEvent(HoverEvent.Action.byName(o.get(0)).buildHoverEvent(Text.of(o.get(1)))), 2, new String[]{"show_entity", "show_item", "show_text"}))
+                .build();
+
+        private final BiFunction<Style, List<String>, Style> styler;
+        private final LiteralText argument;
+        private final List<String> optional;
+
+        public FormattedText(BiFunction<Style, List<String>, Style> styler, LiteralText argument, List<String> optional) {
+            this.styler = styler;
+            this.argument = argument;
+            this.optional = optional;
+        }
+
+        public LiteralText style() throws CommandSyntaxException {
+            try {
+                return (LiteralText) this.argument.setStyle(this.styler.apply(this.argument.getStyle(), this.optional));
+            } catch (Exception e) {
```

Comment:
What Exception are you expecting here?

Reply:
Ah, forgot to remove that when I changed the method of applying styling.